app/Http/Controllers/API/MaterialApiController.php

```diff
@@ -27,8 +27,8 @@ public function index()
 
     public function getByCategory($categoryId){
         //return all data of materials by category id
-        $categories = Category::find($categoryId);
-        return $categories->materials ;
+        $categories = Category::find($categoryId)->materials->skip(0)->take(10);
```

Comment:
you should get offset and limit from the app request, (get it using $request=app('request'))